docs/apidoc/index.rst

```diff
@@ -44,6 +44,17 @@ API Reference
    transpiler_builtin_plugins
    utils
    utils_mitigation
-   opflow
-   algorithms
    exceptions
+
+Deprecated Modules
+==================
+
+.. warning::
+
+   These modules are going to be removed in Qiskit 1.0. Consider pinning ``qiskit~=0.45`` in your dependencies if you need them.
+
+.. toctree::
+   :maxdepth: 1
+
+   algorithms
+   opflow
```

Comment:
You might want this section heading to use a `----` underline, so that it appears as a subsection of "API Documentation", rather than a heading of the same importance?  It's worth checking if that builds correctly with respect to the Furo sidebar, though - I don't know what it would do.

Reply:
> It's worth checking if that builds correctly with respect to the Furo sidebar, though - I don't know what it would do.

Should be fine. The Furo sidebar shows every HTML page nested under the current one. We could remove the `Deprecated Modules` header and the left sidebar would be the same.